backend/kernelCI_app/management/commands/helpers/kcidbng_ingester.py

```diff
@@ -35,11 +38,40 @@
 
 logger = logging.getLogger("ingester")
 
-# Thread-safe queue for database operations
-db_queue = Queue()
+# Batching and backpressure controls
+try:
+    INGEST_BATCH_SIZE = int(os.environ.get("INGEST_BATCH_SIZE", "1000"))
+except Exception:
+    INGEST_BATCH_SIZE = 10000
+
+try:
+    INGEST_FLUSH_TIMEOUT_SEC = float(os.environ.get("INGEST_FLUSH_TIMEOUT_SEC", "2.0"))
```

Comment:
It might be a nice idea to move these into `settings.py`, to rely on proper `from django.conf import settings` handling, even if we create a `ingester_settings.py` that is included into the main settings file... but I can see this as a future refinement.

Reply:
Agree